OutOfSchool/OutOfSchool.DataAccess/Repository/AchievementRepository.cs

```diff
@@ -77,7 +77,15 @@ public async Task<Achievement> Update(Achievement achievement, List<Guid> childr
 
         newAchievement.Teachers.RemoveAll(x => !teachers.Contains(x.Title));
 
+        var deletedTeachers = newAchievement.Teachers.Where(x => teachers.Contains(x.Title) && x.IsDeleted);
+
+        foreach (var teacher in deletedTeachers)
+        {
+            teacher.IsDeleted = false;
+        }
+
```

Comment:
(Minor issue - naming) `deletedTeachers` could be better named `teachersToRestore` or `...Undelete`. Because currently it looks strange, as code set `IsDeleted = false` for deleted entities.

(Major isue - tests) Please cover your changes by unti-test

Reply:
Done